File: 3rd Semester/Advanced Programming Methods/labs/9th lab/model/AbstractContainer.cs
namespace Seminar_10;

public abstract class AbstractContainer : IContainer
{
    protected List<Task> tasks = new List<Task>();

    public abstract Task Remove();

    public abstract void Add(Task task);

    public int Size()
    {
        return tasks.Count;
    }

    public bool IsEmpty()
    {
        return tasks.Count == 0;
    }
}

File: 3rd Semester/Advanced Programming Methods/labs/2nd lab/model/AbstractContainer.java
package sem1_2.model;

import java.util.ArrayList;
import java.util.List;

public abstract class AbstractContainer implements Container{
    protected List<Task> tasks = new ArrayList<>();

    public abstract Task remove();

    public abstract void add(Task task);

    @Override
    public int size() {
        return tasks.size();
    }

    @Override
    public boolean isEmpty() {
        return tasks.isEmpty();
    }
}
